src/views/PeopleSearch/index.js

```diff
@@ -426,39 +430,42 @@ class PeopleSearch extends Component {
           homeCity: this.state.searchValues.homeCity?.trim(),
         },
       });
-      let peopleSearchResults = await goStalk.search(...Object.values(this.state.searchValues));
-
-      if (peopleSearchResults.length === 0) {
+      let results = await goStalk.search(...Object.values(this.state.searchValues));
+      if (results.length === 0) {
         this.setState({
-          peopleSearchResults: noResultsCard,
+          resultData: noResultsCard,
           header: '',
         });
       } else {
         this.setState({
-          peopleSearchResults: (
+          header: (
             <Media query="(min-width: 960px)">
               {(matches) =>
-                peopleSearchResults.map((person) => (
+                matches && !this.state.displayLargeImage
+                  ? peopleSearchHeaderDesktop
+                  : peopleSearchHeaderMobile
+              }
+            </Media>
+          ),
+        });
```

Comment:
Good question. The original writer of the code didn't seem to understand that there was a large difference between having one `setState` with two properties set in it and having two `setState`'s with one property set each. They are split up now, because the output before really wasn't desirable. It has something to do with using the `media` tag, I think, because it doesn't really behave the same way as a lot of other tags. 

The reason header is now _first_ is because, well, setting the header state is a lot faster and smaller, and it is higher up on the screen.

Reply:
I think I can answer the second question. The header is what displays the loading state while loading search results (which may not be the ideal way to do it, but is the current situation nonetheless). Previous developers made a habit of using state to store JSX, rather than keeping simple variables (like a `loading` boolean) in state and handling the JSX based on state in the render method. There are certainly tradeoffs that could be discussed either way, but the general convention among react developers (and within our codebase) is keep state data-focused and handle the logic for rendering that data as JSX within the render method.